llvm/lib/IR/RepoTicket.cpp

```diff
@@ -139,12 +139,25 @@ static bool updateDigestUseDependenciesAndContributions(
   GOHash.update('T');
   auto State = AccumulateGODigest(GO, GOHash, GOIMap);
   Changed = Changed || State.Changed;
-  // Update the GO's hash using its contributions and dependencies.
-  auto &GOContributions = State.Contributions;
-  auto &GODependencies = State.Dependencies;
-  for (const GlobalObject *const G : llvm::concat<const GlobalObject *const>(
-           make_range(GOContributions.begin(), GOContributions.end()),
-           make_range(GODependencies.begin(), GODependencies.end()))) {
+  // The hashes of the GO's 'contributions' and 'dependencies' are separately
+  // added to the GO's hash since their respective arrows point are in different
+  // direction. This results in separating the 'visited' table for the two
+  // directions. In other words, the 'contributions' get a visited map and the
+  // 'dependencies' get a separate map. However, 'contributions' cannot form
+  // loops since 1) only global variables have 'contributions' and 2) all
+  // 'contributions' are functions. Therefore, the 'contributions' can be
+  // simplified not to record 'visited' table, which avoids having two 'visited'
+  // maps.
+  // Add the GO's contributions to its hash.
```

Comment:
This comment left over from the previous version seems rather incongruous. It doesn’t say much more than the code on line 157.

Reply:
I am trying to put two similar comments just before two loops. One is for contribution, and another is for dependencies on line 157. I modified the comment. Is it OK?